test/unit/listeners/substrate-listener/commands/service-request/service-request-created/service-request-created.handler.spec.ts

```diff
@@ -0,0 +1,147 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { countryServiceMockFactory, mailerManagerMockFactory, MockType, stateServiceMockFactory, transactionLoggingServiceMockFactory } from "../../../../../mock";
+import { MailerManager, ProcessEnvProxy, RequestStatus, TransactionLoggingService } from "../../../../../../../src/common";
+import { ServiceRequestCreatedHandler } from "../../../../../../../src/listeners/substrate-listener/commands/service-request/service-request-created/service-request-created.handler";
+import { CountryService } from "../../../../../../../src/endpoints/location/country.service";
+import { StateService } from "../../../../../../../src/endpoints/location/state.service";
+import { BlockMetaData } from "../../../../../../../src/listeners/substrate-listener/models/block-metadata.event-model";
+import { ServiceRequestCreatedCommand } from "../../../../../../../src/listeners/substrate-listener/commands/service-request";
+import { when } from "jest-when";
+import * as rewardCommand from "../../../../../../../src/common/polkadot-provider/command/rewards";
+import { TransactionLoggingDto } from "../../../../../../../src/common/modules/transaction-logging/dto/transaction-logging.dto";
+
+describe('Service Request Created Handler Event', () => {
+  let serviceRequesCreatedHandler: ServiceRequestCreatedHandler;
+  let transactionLoggingServiceMock: MockType<TransactionLoggingService>;
+  let countryServiceMock: MockType<CountryService>;
+  let stateServiceMock: MockType<StateService>;
+  let mailerManagerMock: MockType<MailerManager>;
+
+  const createMockRequest = (requestStatus: RequestStatus) => {
+    return [
+      {},
+      {
+        toHuman: jest.fn(
+          () => ({
+            hash: "string",
+            requesterAddress: "string",
+            labAddress: "string",
+            country: "XX",
+            region: "XX",
+            city: "XX",
+            serviceCategory: "Test",
+            stakingAmount: "1000000000000",
+            status: requestStatus,
+            createdAt: new Date(),
+            updatedAt: new Date(),
+            unstakedAt: new Date()
+          })
+        )
+      }
+    ]
+  }
+
+	function mockBlockNumber(): BlockMetaData {
+		return {
+			blockHash: "string",
+			blockNumber: 1,
+		}
+	}
+
+  const EMAILS = "email";
+  
+  class ProcessEnvProxyMock {
+    env = { EMAILS };
+  }
+
+  const NOW = new Date().toString();
+
+  jest.spyOn(global, 'Date').mockImplementationOnce(() => NOW);
```

Comment:
Where is `Date.now` used? Instead of mocking the global `Date` please refactor the file to use `DateTimeProxy` instead and mock that.